torch/autograd/_functions/linalg.py

```diff
@@ -102,3 +102,44 @@ def backward(ctx, grad_output, grad_LU=None):
         grad_b, _ = torch.gesv(grad_output, a.t())
         grad_a = -torch.mm(grad_b, X.t())
         return grad_b, grad_a
+
+class Potrf(Function):
+    """
+    cf. Iain Murray (2016); arXiv 1602.07527
+    """
+
+    @staticmethod
+    def forward(ctx, a, upper=True):
+        ctx.upper = upper
+        fact = torch.potrf(a, upper)
+        ctx.save_for_backward(fact)
+        return fact
+
+    @staticmethod
+    def Phi(A):
+        """
+        Return lower triangle of A and halve the diagonal.
+        """
+        B = A.tril()
+        for i in range(B.size(0)):
+            B[i, i] = B[i, i] * 0.5
```

Comment:
So, it turns out that there is not yet a version of `torch.eye` that works directly on the GPU.
I implemented it in https://github.com/pytorch/pytorch/pull/2395 , so once it's merged, you could do something like
```python
mat = torch.eye(A.size(0), out=A.data.new()) * (-0.5) + 1
```